### ISSUE TITLE
Add validity checks for OpDecorate and OpMemberDecorate in validator

### ISSUE THREAD

**Issue by scyganINTEL** [KhronosGroup/SPIRV-Tools#207]
There is a set of non-trivial constraints in SPIR-V & Vulkan specs regarding decorating variables with: 

- location & binding qualifiers
- interpolation and patch qualifiers
- offset and alignment 

Especially nested interface blocks /  structs  / arrays need to follow rules defined in "14.1. Shader Input and Output Interfaces" Vulkan spec chapter to get SPIR-V shaders properly "linked" during pipeline creation. 

As there are apps (including glslang -  https://github.com/KhronosGroup/glslang/pull/269) misusing the decorations it would be good to have these checked in SPIR-V validator.

**Comment by dneto0** [comment]
Agreed.  @umar456 is working on the validator.  I believe no Vulkan-specific checks have been implemented yet.

**Comment by prideout** [comment]
Note that `spvValidate` currently operates on a single binary, so I think this issue implies that it should actually take a list of binaries in order to validate consistent decorations, e.g. the outs of a VS should match the ins of a FS.
